### PR TITLE
Add example of higher-order effect handler (LocalReader)

### PR DESCRIPTION
**Pull request by Lysxia**
If you use the existing way of defining custom effects outlined in `Compound` and try to add operations that take `Eff` computations as arguments... it just works.

### CODE REVIEW

File: bluefin-internal/src/Bluefin/Internal/Examples.hs [hunk 1]
```diff
@@ -1021,3 +1021,23 @@ rethrowIOExample = runEff $ \io -> do
   effIO io $ putStrLn $ case r of
     Left e -> "Caught IOException:\n" ++ show e
     Right contents -> contents
+
+data LocalReader r e = LocalReader
+  { askLR :: forall es. Eff (es :& e) r
+  , localLR :: forall es a. (r -> r) -> Eff (es :& e) a -> Eff (es :& e) a
```
Comment (tomjaguarpaw) [inline bluefin-internal/src/Bluefin/Internal/Examples.hs:1027]: I'm _very_ surprised that `e` can appear in negative position here, yet not prevent `mapHandle` from being defined. Is my intuition completely wrong?

Comment (Lysxia) [reply bluefin-internal/src/Bluefin/Internal/Examples.hs:1027]: I also didn't think it was possible until just now. This type is equivalent to `forall es a. e :> es => (r -> r) -> Eff es a -> Eff es a` which has a more obvious polarity.

Comment (tomjaguarpaw) [reply bluefin-internal/src/Bluefin/Internal/Examples.hs:1027]: Wow, my intuition really needs upgrading.

**Comment by tomjaguarpaw** [comment]
Nice! Thanks for this. It would make a good example for a new "Dynamic effects with effectful operations as arguments" section for `Bluefin.Compound`. I need to prioritise defining the necessary machinery for `interpose`, but once I've done that I'll come back to this.

**Comment by tomjaguarpaw** [comment]
Thanks, I slightly edited this and incorporated it as https://github.com/tomjaguarpaw/bluefin/commit/eec720c8b3e6c9f26d537b1fc8729dd249383241. I also incorporated it into the docs, due to appear in https://hackage.haskell.org/package/bluefin-0.0.12.1/docs/Bluefin-Compound.html

**Comment by Lysxia** [comment]
Cheers!

**Pull request closed without merging**